src/openmrs.ts

```diff
@@ -1,28 +1,52 @@
-import { setPublicPath } from 'systemjs-webpack-interop';
+import { setPublicPath } from "systemjs-webpack-interop";
+import { messageOmrsServiceWorker, subscribeNetworkRequestFailed } from "@openmrs/esm-framework";
 
-const moduleName = '@openmrs/esm-form-entry-app';
-
-setPublicPath(moduleName);
-
-const backendDependencies = {
-  'webservices.rest': '2.24.0',
-};
+setPublicPath("@openmrs/esm-form-entry-app");
 
+const backendDependencies = { "webservices.rest": "2.24.0" };
 const importTranslation = require.context(
-  '../translations',
+  "../translations",
   false,
   /.json$/,
-  'lazy'
+  "lazy"
 );
 
 function setupOpenMRS() {
+  messageOmrsServiceWorker({
+    "type": "registerDynamicRoute",
+    "pattern": ".+/visit.+"
+  });
+
+  messageOmrsServiceWorker({
+    "type": "registerDynamicRoute",
+    "pattern": ".+/ws/fhir2/R4/Observation.+"
+  });
+
+  messageOmrsServiceWorker({
+    "type": "registerDynamicRoute",
+    "pattern": ".+/ws/rest/v1/obs.+"
+  });
+
+  messageOmrsServiceWorker({
+    "type": "registerDynamicRoute",
+    "pattern": ".+/ws/rest/v1/appui/session.*"
+  });
+
+  subscribeNetworkRequestFailed(({request}) => {
+    console.log('Request failed: ', request);
```

Comment:
I guess the `console.log` is temporary here, right?

Reply:
Yep, will remove it.